src/jit/emitarm64.cpp

```diff
@@ -10860,15 +10858,37 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
             {
                 assert(genCountBits(dst->gtRsvdRegs) == 1);
                 extraReg = genRegNumFromMask(dst->gtRsvdRegs);
+                bool shouldEmitMultiplication = true;
+                if (attr == EA_4BYTE)
+                {
+                    // Compute 8 byte results from 4 byte by 4 byte multiplication.
+                    emitIns_R_R_R(INS_umull, EA_8BYTE, dst->gtRegNum, src1->gtRegNum, src2->gtRegNum);
+
+                    // Get the high result by shifting dst.
+                    emitIns_R_R_I(INS_lsr, EA_8BYTE, extraReg, dst->gtRegNum, 32);
 
-                // Compute the high result
-                emitIns_R_R_R(ins2, attr, extraReg, src1->gtRegNum, src2->gtRegNum);
+                    if (extraReg != dst->gtRegNum)
```

Comment:
There is a way to insure that extraReg will never be dst->gtRegNum,
I think it involves asking for two extra registers in lower.
We used that in a couple of places already.